File: backend/app/schemas/patient.py
from typing import Optional

from pydantic import BaseModel, EmailStr, ConfigDict


class PatientCreate(BaseModel):
    first_name: str
    last_name: str
    age: int
    gender: str
    disease: Optional[str] = None
    phone: Optional[str] = None
    blood_group: Optional[str] = None
    email: Optional[EmailStr] = None
    height: Optional[float] = None


class PatientUpdate(BaseModel):
    first_name: Optional[str] = None
    last_name: Optional[str] = None
    age: Optional[int] = None
    gender: Optional[str] = None
    disease: Optional[str] = None
    phone: Optional[str] = None
    blood_group: Optional[str] = None
    email: Optional[EmailStr] = None
    height: Optional[float] = None


class PatientRead(BaseModel):
    id: int
    first_name: str
    last_name: str
    age: int
    gender: str
    disease: Optional[str]
    phone: Optional[str]
    blood_group: Optional[str]
    email: Optional[EmailStr]
    height: Optional[float]

    model_config = ConfigDict(from_attributes=True)

File: frontend/lib/patient.ts
import api from "./api";

export interface Patient {
  id?: number;
  first_name: string;
  last_name: string;
  age: number;
  gender: string;
  disease?: string;
  phone?: string;
  blood_group?: string;
  email?: string;
  height?: number;
}

export async function getPatients(
  token: string,
  search = ""
) {
  const response = await api.get("/patients", {
    params: {
      search,
    },
    headers: {
      Authorization: `Bearer ${token}`,
    },
  });

  return response.data;
}

export async function createPatient(
  patient: Patient,
  token: string
) {
  const response = await api.post(
    "/patients",
    patient,
    {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    }
  );

  return response.data;
}

export async function updatePatient(
  id: number,
  patient: Patient,
  token: string
) {
  const response = await api.put(
    `/patients/${id}`,
    patient,
    {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    }
  );

  return response.data;
}

export async function deletePatient(
  id: number,
  token: string
) {
  await api.delete(`/patients/${id}`, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
  });
}